src/threading.c

```diff
@@ -724,16 +728,57 @@ JL_DLLEXPORT void jl_exit_threaded_region(void)
     }
 }
 
+// Profiling stubs
+
+#ifdef USE_ITTAPI
+
+#define profile_lock_init(lock, name) __itt_sync_create(lock, "jl_mutex_t", name, __itt_attr_mutex)
```

Comment:
Would it be possible to move these definitions into `timing.c/.h`?

Ideally this would follow the conventions established there to provide backend-specific implementations (in particular, it should be possible to turn on multiple back-end implementations at once)

Reply:
I'm also OK if these stay local to `threading.c`, but I would like to try to support multiple back-ends being enabled at once

Reply:
Would a static inline function fix that problem?

Reply:
Yeah... Let's go with the static_inline